Hamburguer menu actions

JuanDiegoGuerra opened the pull request:
Create "app.js"
 -
> Implemented mobile - menu (Pair Programming): ft. @CesarHerr  🤝🏼
 - Menu Icon (hamburguer icon) (javascript) ✅
 - Smooth scrolling (css transition) ✅
> Changes Made 🛠 : 
 - Created app.js 🔨
 - Updated index.html 🔧
    - Added "Home" link in navbar.
 - Updated mobile_first.css 🔧
    - Added styles for hamb menu.
 - Updated animation_transitions.css 🔧
    - Fixed navbar's listed items transitions.

> Sumary Preview : 🕵🏼‍♂️
- Mobile Menu
![newnavbar](https://user-images.githubusercontent.com/45726631/231044571-2d07dfd3-1a02-4c66-ad5b-149f38d35c82.gif)

- Smooth Scrolling
![smoothscrollingbar2](https://user-images.githubusercontent.com/45726631/231045098-7d746453-e470-459f-bc45-1e81fb9db187.gif)

JuanDiegoGuerra committed:
Update index.html

Added anker links

JuanDiegoGuerra committed:
Update mobile_first.css

Added all the styles for the new navbar

CesarHerr committed:
Update index.html

Add app.js link reference

CesarHerr committed:
Update app.js

Add constants to app.js file

CesarHerr committed:
Update app.js

Added event listeners

JuanDiegoGuerra committed:
Update index.html

Added new listed item "Home" for navbar

JuanDiegoGuerra committed:
Update animations_transitions.css

Fixed "listed-item" transition from navbar in order to transform for bigger mobile sizes

JuanDiegoGuerra committed:
Update mobile_first.css

Added some styles and effects for navbar's "listed-items"

JuanDiegoGuerra committed:
Update mobile_first.css

Added a smoothie scroll behavior for the anchor links of the navbar

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed Linters

JuanDiegoGuerra committed:
Update animations_transitions.css

Fixed Linters

JuanDiegoGuerra committed:
Modified Eslint file

JuanDiegoGuerra committed:
Update app.js

Fixed Linter

ABDULALI3468 commented on css/mobile_first.css:
```diff
@@ -618,10 +622,65 @@ footer h3:hover {
   }
 
   .flex-nav-list {
-    display: none;
+    font-size: 15px;
+    display: flex;
+    position: fixed;
+    left: -150%;
+    top: -20px;
+    padding-top: 10px;
+    gap: 0;
+    background-color: var(--verde);
+    border: 3px black solid;
+    width: 100%;
+    text-align: center;
+    transition: 0.5s;
+    font-weight: 700;
+  }
+
```
Please consider using the same set of styles for the navbar as used in the Figma template as it's a project requirement and to make your project more professional.

## Your Design:-

![image](https://user-images.githubusercontent.com/94223367/231049987-e1025a7f-c8e6-4353-b5b4-61e3c3eed712.png)


## Figma Design:-

![image](https://user-images.githubusercontent.com/94223367/231049922-7b04af0d-5323-46fd-9dcf-5f0d267db842.png)

JuanDiegoGuerra committed:
Update README.md

Adding a contributor of the project :)

ABDULALI3468 reviewed (changes requested):
# Changes Required 👍 

Hi @JuanDiegoGuerra @CesarHerr , 

<img src="https://media1.giphy.com/media/31Q36EHJ7C5ZoIMw79/giphy.gif"/>

Indeed you did an exceptional but still, there is room for changes to complete this project. 💯


## Highlights 
✅Navbar is working smoothly.
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.


 ## Required Changes ♻️

> **1-** Please consider removing the horizontal scroll behavior from your portfolio as its kind of makes the user experience bad and somewhat irritating as the user has to scroll to the right to see the other content.

## Optional suggestions

**[OPTIONAL]** Please consider using the same set of styles for the complete page as used in the Figma template to make your project more professional. Remember it's optional to do as it's not asked in this project requirement but your styles are not in accordance with the design.

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.

> * This is My **GitHub**: ![GitHub followers](https://img.shields.io/github/followers/ABDULALI3468?label=ABDULALI&style=social)
> * This is My **Linkedin**: [![Linkedin Badge](https://img.shields.io/badge/-ABDUL%20ALI-blue?style=flat-square&logo=Linkedin&logoColor=white&link=https://www.linkedin.com/in/abdul-ali-5400bb216/)](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [![Twitter Badge](https://img.shields.io/badge/-@mrabdul_ali_-1ca0f1?style=flat-square&labelColor=1ca0f1&logo=twitter&logoColor=white&link=https://twitter.com/mrabdul_ali)](https://twitter.com/mrabdul_ali)&nbsp;

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

JuanDiegoGuerra committed:
Update index.html

added an id for Home section

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed multiples element's styles in order to stick to the... design

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed linter erros

JuanDiegoGuerra commented:
Updated index.html & mobile_first.css
 -
> Made the Required Changes ♻.
 - Display navbar items in a column ✅.
 - Using palete of colors from the design ✅.
 - Added box shadow for the bottom part of the menu ✅.
 - Fixed elements position ✅.
> Sumary Preview 🕵🏼‍♂️ :

![navbarrequierdchanges3](https://user-images.githubusercontent.com/45726631/231090740-91221ce9-2032-4e42-8fae-19ac4e129a08.gif)

aamirkhan2478 commented on js/app.js:
```diff
@@ -0,0 +1,12 @@
+const hamburger = document.querySelector('.menu');
+const navMenu = document.querySelector('.flex-nav-list');
+
+hamburger.addEventListener('click', () => {
+  hamburger.classList.toggle('active');
+  navMenu.classList.toggle('active');
+});
+
+navMenu.addEventListener('click', () => {
+  hamburger.classList.toggle('active');
+  navMenu.classList.toggle('active');
+});
```
[OPTIONAL] Please use the `DRY` principle your code is not `DRY`. `DRY` stand for Don't Repeat Yourself, a basic principle of software development aimed at reducing the repetition of information. This requirement is optional because the previous reviewer miss this requirement so kindly don't skill this requirement because this recommended requirement :slightly_smiling_face: check below :point_down: my suggestion :+1: 

```suggestion
const menuButton = document.querySelector('.menu');
const navMenu = document.querySelector('.flex-nav-list');

const toggleMenu = () => {
  menuButton.classList.toggle('active');
  navMenu.classList.toggle('active');
}

menuButton.addEventListener('click', toggleMenu);
navMenu.addEventListener('click', toggleMenu);
```

- The first change is to rename the hamburger variable to menuButton to make the purpose of the variable more clear. Additionally, we can extract the anonymous function that toggles the "active" class on both elements into a named function toggleMenu() to make the code more readable.

- We can then pass this named function as the event listener to both the menuButton and navMenu elements.

- Finally, we remove the arrow function from the event listeners and simply pass the toggleMenu function as the argument. This makes the code more concise and easier to read.

aamirkhan2478 reviewed (approved):
Hi @JuanDiegoGuerra and @CesarHerr , 

![GIF](https://www.funimada.com/assets/images/cards/big/congrats-7.gif)

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly
✅ No linter error
✅ Good commit messages
✅ Good descriptive PR

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
